Sources/MessagingPush/MessagingPush.swift

```diff
@@ -69,19 +85,29 @@ public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, Messagi
         writeKey: String,
         configure configureHandler: ((inout MessagingPushConfigOptions) -> Void)? = nil
     ) -> MessagingPushInstance {
-        initialize { config in
-            config.writeKey = writeKey
-            if let configureHandler = configureHandler {
-                configureHandler(&config)
-            }
+        if let configureHandler = configureHandler {
+            configureHandler(&moduleConfig)
         }
+        moduleConfig.writeKey = writeKey
+
+        shared.initializeModuleIfNotAlready()
+
+        return shared
     }
 
-    private func initializeModule() {
-        guard getImplementationInstance() == nil else {
```

Comment:
`main` contains new integration tests that verifies the push module only gets initialized once. That test failed with the current implementation. I replaced the implementation with the one in `main` and the test passed.

Reply:
Do you know the cause of its failure? Is the behavior change related to `lock` or `@Atomic` variable check? I’m asking because we have similar implementation in other modules, and if there’s a common issue, we should fix it before final release.

No action needed for this PR though, we can address this in a separate ticket.

Reply:
Made a [ticket](https://linear.app/customerio/issue/MBL-129/sdk-modules-can-only-be-initialized-1-time) for this. 

I do not remember why `guard getImplementationInstance() == nil else {` does not work and a mutex lock does. Luckily, I was able to write an automated test that reproduced the bug and the mutex lock made the test pass. So, I feel more confident that we have a working solution after being able to reproduce the behavior.